aws/cloudformation/components/ami.yml.erb

```diff
@@ -163,8 +163,18 @@ frontend_device_name ||= '/dev/sda1'
 
           STACK=${AWS::StackName}
           REGION=${AWS::Region}
+          AUTO_SCALING_GROUP=Frontends-${AWS::StackName}
+          ENVIRONMENT=<%=environment%>
```

Comment:
I'm copying how the ENVIRONMENT env var is set in bootstrap_chef_stack.sh.erb: https://github.com/code-dot-org/code-dot-org/blob/697167311a5338bf7d443c0ac691f581fe06968c/aws/cloudformation/bootstrap_chef_stack.sh.erb/#L24

In cloud_formation_stack.yml.erb, it directly assumes that $ENVIRONMENT is already set https://github.com/code-dot-org/code-dot-org/blob/697167311a5338bf7d443c0ac691f581fe06968c/aws/cloudformation/cloud_formation_stack.yml.erb/#L606

Maybe this line isn't required? No idea how to test, so I went with what seemed most conservative.

Reply:
This file, "ami.yml.erb" is included in "cloud_formation_stack" yml, so the variable should be accessible here. `ENVIRONMENT=<%=$ENVIRONMENT=>` should work. Or just use `$ENVIRONMENT` below, though I prefer getting out an ERB context as soon as possible by assigning some variables as you did.

Since this is ERB, I think you could test by validating it locally:

`bundle exec rake stack:validate RAILS_ENV=production`

You might need to set `VERBOSE=1` first to get the template to print.